features/305.Big_Data_Spark.feature

```diff
@@ -0,0 +1,133 @@
+        And   I wait "2" seconds
+
+    # -- Subscribing to Context changes. Request 1
+    Scenario: Request 1 - Create a subscription with the Context Broker
+      When  I prepare a POST HTTP request to "http://localhost:1026/v2/subscriptions"
+      And   I set header fiware-service to openiot
+      And   I set header fiware-servicepath to /
+      And   I set the body request as described in 01.request_305.json
+      And   I perform the request
+      Then  I receive a HTTP response with status 201 and empty dict
+
+    # The next Scenario outline is not shown in the tutorial, but it is writen textually:
+    #
+    # "... if this is not the case, then the devices are not regularly sending data. \
+    # Remember to unlock the Smart Door and switch on the Smart Lamp"
+    #
+    # so I translate English to Context-Broker
+    Scenario Outline: Pre-Request 2 scenario - Do something with the devices
+      When  I prepare a PATCH HTTP request to "http://localhost:1026/v2/entities/<device>/attrs"
+      And   I set header fiware-service to openiot
+      And   I set header fiware-servicepath to /
+      And   I set the body request as described in <request_file>
+      And   I perform the request
+      Then  I receive a HTTP response with status <response_code> and empty dict
+      Examples:
+        | request_file           | device     | response_code  |
+        | ring_bell_req.json     | Bell:001   | 204            |
+        | unlock_door.json       | Door:001   | 204            |
+
+    # Check the subscrition is done.
+    Scenario: Request 2 - Test the subscriptions
+       When  I prepare a GET HTTP request to "http://localhost:1026/v2/subscriptions/"
+       And   I set header fiware-service to openiot
+       And   I set header fiware-servicepath to /
+       And   I perform the query request
+       Then  I receive a HTTP "200" response code from Broker with the body "02.response_305.json" and exclusions "02.excludes"
+
+    #  Remember to unlock the Smart Door and switch on the Smart Lamp
+    # So I added a couple things before Request 2 to fulfill that requirement.
+    #
+    Scenario: Logger - checking the output
+      Given I wait "60" seconds
+      When  I Compare next lines in terminal spark-run-jar at least I can find 1 in stdout with a timeout 10 matching filename 01.expected_on_terminal.txt
+      And   I flush the terminal spark-run-jar queues
+
+    # LOGGER NGSI-DL:
+    # I guess this should be the same as previous case but for LD. Don't know if I have to do something in
+    # between for this to work.
+    Scenario: Run the generated jar in Spark using the LoggerLD. This process won't end
+      Given I kill process in terminal spark-run-jar
+      And   I wait "10" seconds
+      Given I prepare the script run-spark-ld-jar.sh to be run
+      When  I open a new shell terminal spark-run-ld-jar and exec "features/data/305.Big_Data_Spark/run-spark-ld-jar.sh"
```

Comment:
When  I open a new shell terminal spark-run-ld-jar and execute "features/data/305.Big_Data_Spark/run-spark-ld-jar.sh"

Reply:
idem

Reply:
NTC